brave-apache-http-interceptors/src/main/java/brave/httpclient4/HttpClient4Instrumentation.java

```diff
@@ -0,0 +1,126 @@
+package brave.httpclient4;
+
+import brave.ClientHandler;
+import brave.Span;
+import brave.Tracer;
+import brave.parser.Parser;
+import brave.parser.TagsParser;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import java.io.IOException;
+import java.net.URI;
+import org.apache.http.HttpMessage;
+import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.HttpResponseInterceptor;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import zipkin.TraceKeys;
+
+import static com.github.kristofa.brave.internal.Util.checkNotNull;
+
+public final class HttpClient4Instrumentation {
```

Comment:
actually this won't work because there's a lot of logic inside the HttpClientBuilder which is only invoked if HttpProcessor isn't overridden. IOTW, I don't think we'll be able to neatly do implicit propagation. Rather, we'd need to document the attribute we use internally if someone wants to access the current span.